src/manager/keyboard.rs

```diff
@@ -1,19 +1,21 @@
 //! Handler for keyboards
 
-use events::key_events::KeyEvent;
 use libc;
-
-use types::device::Device;
+use events::key_events::KeyEvent;
+use types::input_device::InputDevice;
+use types::keyboard::Keyboard;
 use wlroots_sys::wlr_event_keyboard_key;
 
 pub trait KeyboardHandler {
     /// Callback that is triggered when a key is pressed.
-    fn on_key(&mut self, &mut Device, &KeyEvent) {}
+    fn on_key(&mut self, KeyEvent) {}
```

Comment:
This should pass a (mutable) reference to the `KeyEvent`. Otherwise user code can copy it into their data structure and use it after the pointer it refers to has been freed.